packages/tools/devtools/devtools-example/src/FluidObject.ts

```diff
@@ -0,0 +1,162 @@
+	 */
+	private readonly sharedTextKey = "shared-text";
+
+	/**
+	 * Key in the app's `rootMap` under which the SharedCounter object is stored.
+	 */
+	private readonly sharedCounterKey = "shared-counter";
+
+	/**
+	 * Key in the app's `rootMap` under which the SharedCell object is stored.
+	 */
+	private readonly emojiMatrixKey = "emoji-matrix";
+
+	/**
+	 * Key in the app's `rootMap` under which the SharedTree object is stored.
+	 */
+	private readonly sharedTreeKey = "shared-tree";
+
+	/**
+	 * Key in the app's `rootMap` under which the SharedDirectory object is stored.
+	 */
+	private readonly initialObjectsDirKey = "rootMap";
+
+	// previous app's `rootMap`
+	private readonly _initialObjects: Record<string, IFluidLoadable> = {};
+
+	private _text: SharedString | undefined;
+	private _counter: SharedCounter | undefined;
+	private _emojiMatrix: SharedMatrix | undefined;
+
+	public get text(): SharedString {
+		if (this._text === undefined) {
+			throw new Error("The SharedString was not initialized correctly");
+		}
+		return this._text;
+	}
+
+	public get counter(): SharedCounter {
+		if (this._counter === undefined) {
+			throw new Error("The SharedCounter was not initialized correctly");
+		}
+		return this._counter;
+	}
+
+	public get emojiMatrix(): SharedMatrix {
+		if (this._emojiMatrix === undefined) {
+			throw new Error("The SharedMatrix was not initialized correctly");
+		}
+		return this._emojiMatrix;
+	}
+
+	public getRootObject(): Record<string, IFluidLoadable> {
+		return this._initialObjects;
+	}
+
+	public static get Name(): string {
+		return "@devtools-example/test-app";
+	}
+
+	private static readonly factory = new DataObjectFactory(
+		CollaborativeDevtoolExampleAppObjects.Name,
+		CollaborativeDevtoolExampleAppObjects,
+		[
+			SharedString.getFactory(),
+			SharedCounter.getFactory(),
+			SharedMatrix.getFactory(),
+			SharedCell.getFactory(),
+			SharedTree.getFactory(),
+		],
+		{},
+	);
+
+	public static getFactory(): DataObjectFactory<CollaborativeDevtoolExampleAppObjects> {
+		return this.factory;
+	}
+
+	protected async initializingFirstTime(): Promise<void> {
+		// Create the SharedString and store the handle in our root SharedDirectory
+		const text = SharedString.create(this.runtime, this.sharedTextKey);
+		const counter = SharedCounter.create(this.runtime, this.sharedCounterKey);
+		const sharedTree = SharedTree.create(this.runtime, this.sharedTreeKey);
+
+		const emojiMatrix = SharedMatrix.create(this.runtime, this.emojiMatrixKey);
+		const matrixDimension = 2; // Height and Width
+		emojiMatrix.insertRows(0, matrixDimension);
+		emojiMatrix.insertCols(0, matrixDimension);
+		for (let row = 0; row < matrixDimension; row++) {
+			for (let col = 0; col < matrixDimension; col++) {
+				const emojiCell = SharedCell.create(this.runtime);
+				emojiMatrix.setCell(row, col, emojiCell.handle);
+			}
+		}
+		this.root.createSubDirectory(this.initialObjectsDirKey);
```

Comment:
Feels like this extra subDirectory is not necessary...? We're putting all the shared objects directly in the root, which seems fine. `getRootObject()` could directly return `this.root`... I think.

Reply:
Tested it. Actually we do need it or it would be "Error: InitialObjects sub-directory was not initialized".